tests/Unit/PrestaShopBundle/Translation/TranslationCatalogueBuilderTest.php

```diff
@@ -0,0 +1,482 @@
+{
+    private const LOCALE = 'en-US';
+
+    private static $defaultTranslations = [
+        'AdminFirstDomain' => [
+            'First Domain First Wording' => 'First Domain First Wording Default Translation',
+            'First Domain Second Wording' => 'First Domain Second Wording Default Translation',
+        ],
+        'AdminSecondDomain' => [
+            'Second Domain First Wording' => 'Second Domain First Wording Default Translation',
+            'Second Domain Second Wording' => 'Second Domain Second Wording Default Translation',
+        ],
+    ];
+    private static $fileTranslatedTranslations = [
+        'AdminFirstDomain' => [
+            'First Domain First Wording' => 'First Domain First Wording File Translation',
+            'First Domain Second Wording' => 'First Domain Second Wording File Translation',
+        ],
+        'AdminSecondDomain' => [
+            'Second Domain First Wording' => 'Second Domain First Wording File Translation',
+        ],
+    ];
+    private static $userTranslatedTranslations = [
+        'AdminFirstDomain' => [
+            'First Domain First Wording' => 'First Domain First Wording User Translation',
+            'First Domain Second Wording' => 'First Domain Second Wording User Translation',
+        ],
+    ];
+
+    /**
+     * @var TranslationCatalogueBuilder
+     */
+    private $translationCatalogueBuilder;
+
+    public function setUp()
+    {
+        $provider = $this->createMock(CatalogueProviderInterface::class);
+
+        // Build Default catalogue
+        $catalogue = new MessageCatalogue(self::LOCALE);
+        foreach (self::$defaultTranslations as $domain => $messages) {
+            $catalogue->add($messages, $domain);
+        }
+        $provider->method('getDefaultCatalogue')->willReturn($catalogue);
+
+        // Build FileTranslated catalogue
+        $catalogue = new MessageCatalogue(self::LOCALE);
+        foreach (self::$fileTranslatedTranslations as $domain => $messages) {
+            $catalogue->add($messages, $domain);
+        }
+        $provider->method('getFileTranslatedCatalogue')->willReturn($catalogue);
+
+        // Build UserTranslated catalogue
+        $catalogue = new MessageCatalogue(self::LOCALE);
+        foreach (self::$userTranslatedTranslations as $domain => $messages) {
+            $catalogue->add($messages, $domain);
+        }
+        $provider->method('getUserTranslatedCatalogue')->willReturn($catalogue);
+
+        // The factory
+        $providerFactory = $this->createMock(CatalogueProviderFactory::class);
+        $providerFactory->method('getProvider')->willReturn($provider);
+
+        $router = $this->createMock(Router::class);
+        $router->method('generate')->willReturn('route');
+
+        $this->translationCatalogueBuilder = new TranslationCatalogueBuilder($providerFactory, new TranslationsTreeBuilder($router));
+    }
+
+    public function testGetDomainCatalogueWrongType()
```

Comment:
```suggestion
    public function testGetDomainCatalogueFailsWhenGivenWrongType()
```
Imagine than in 1 year, you need to rework these classes and (following your refactoring), your tests fail... at this moment, you will be very happy that you wrote overly explicit method names 😄